suite-native/app/e2e/tests/deeplinkPopup.test.ts

```diff
@@ -0,0 +1,162 @@
+import http from 'http';
+import { exec } from 'child_process';
+
+import { TrezorUserEnvLink } from '@trezor/trezor-user-env-link';
+import TrezorConnect from '@trezor/connect-mobile';
+
+import {
+    appIsFullyLoaded,
+    disconnectTrezorUserEnv,
+    openApp,
+    prepareTrezorEmulator,
+    restartApp,
+} from '../utils';
+import { onOnboarding } from '../pageObjects/onboardingActions';
+import { onCoinEnablingInit } from '../pageObjects/coinEnablingActions';
+import { onHome } from '../pageObjects/homeActions';
+
+const SERVER_PORT = 8080;
+const SERVER_URL = `http://localhost:${SERVER_PORT}`;
+
+let server: http.Server | undefined;
+
+const openUriScheme = (url: string, platformToOpen: 'android') => {
+    const command = `npx uri-scheme open '${url.replace(/'/g, '')}' --${platformToOpen}`;
+
+    exec(command, (err, stdout, stderr) => {
+        if (err) {
+            console.error(err);
+
+            return;
+        }
+        // eslint-disable-next-line no-console
+        console.info(stdout);
+        console.error(stderr);
+    });
+};
+
+const reverseServerPort = (port: number) => {
+    return new Promise((resolve, reject) => {
+        exec(
+            `/usr/local/bin/platform-tools/adb reverse tcp:${port} tcp:${port}`,
+            (err, stdout, stderr) => {
+                if (err) {
+                    console.error(err);
+
+                    return reject(err);
+                }
+                // eslint-disable-next-line no-console
+                console.info(stdout);
+                console.error(stderr);
+                resolve(stdout);
+            },
+        );
+    });
+};
+
+describe('Deeplink connect popup.', () => {
+    beforeAll(async () => {
+        server = http.createServer((req, res) => {
+            if (req.url) {
+                const url = new URL(req.url, SERVER_URL);
+                TrezorConnect.handleDeeplink(url.href);
+                res.statusCode = 200;
+                res.setHeader('Content-Type', 'text/plain');
+                res.end('Callback URL received successfully!\n');
+            }
+        });
+
+        server.listen(SERVER_PORT, 'localhost', () => {
+            // eslint-disable-next-line no-console
+            console.info(`Server running at ${SERVER_URL}`);
+        });
```

Comment:
nitpick: execution doesn't wait for server to be already listening. it could lead to a racecondition that is very unlikely to happen

Reply:
That is a very good point. This should love that https://github.com/trezor/trezor-suite/pull/14667/commits/9b9116f4048b692faf897efdda225a05ab936f31